config/hooks.go

```diff
@@ -4,21 +4,21 @@ import (
 	"github.com/canonical/microcluster/internal/state"
 )
 
-// Hooks is an interface representing customizable functions that can be called at varying points by the daemon to
+// Hooks holds customizable functions that can be called at varying points by the daemon to.
 // integrate with other tools.
-type Hooks interface {
+type Hooks struct {
 	// OnBootstrapHook is run after the daemon is initialized and bootstrapped.
-	OnBootstrapHook(s *state.State) error
+	OnBootstrapHook func(s *state.State) error
```

Comment:
The "Hook" at the end of each field name seems redundant seeing as they belong to the Hooks struct.